src/gwt/src/org/rstudio/studio/client/common/spelling/TypoSpellChecker.java

```diff
@@ -347,6 +363,27 @@ public void prefetchWords(ArrayList<String> words)
       }
    }
 
+   public boolean shouldCheckSpelling(DocDisplay dd, Range r)
+   {
+      String word = dd.getTextForRange(r);
+      // Don't worry about pathologically long words
+      if (r.getEnd().getColumn() - r.getStart().getColumn() > 250)
+         return false;
+
+      // Don't spellcheck yaml
+      Scope s = ((AceEditor)dd).getScopeAtPosition(r.getStart());
+      if (s.isYaml())
+         return false;
+
+      // Don't spellcheck untokenized words inside a keyword scope
+      String line = dd.getLine(r.getStart().getRow());
+      Pattern p =  Pattern.create("\\\\(?:documentclass|usepackage|newcounter|setcounter|addtocounter|value|arabic|stepcounter|newenvironment|renewenvironment|ref|vref|eqref|pageref|label|cite[a-zA-Z]*|tag|begin|end|bibitem).*{.*" + word + ".*}");
```

Comment:
Is there a reasonable way we could pull this pattern out of the highlight rules instead of duplicating it?

Also not clear on how this keyword scope list applies here -- this is for TeX and (a) should not apply to other modes and (b) won't capture keywords in more common types such as Roxygen (e.g. `\code`)

Reply:
Looking at the highlight rules they're just in a big array but maybe I can pull out what we want without too much mess, there appears to be another keyword pattern that is more generic. I was just working with the explicit examples that Mine had given me and that pattern encompassed all of them.